quickwit/quickwit-indexing/src/actors/merge_executor.rs

```diff
@@ -375,7 +375,7 @@ impl MergeExecutor {
     async fn process_delete_and_merge(
         &mut self,
         merge_split_id: String,
-        mut splits: Vec<SplitMetadata>,
+        splits: Vec<SplitMetadata>,
```

Comment:
If we only accept splits.len() == 1, maybe we would be better off calling this `process_delete_and_compact` and passing  `split: SplitMetadata` as argument.

Reply:
(this bug would have been avoided.)